include/libopencm3/stm32/mp1/memorymap.h

```diff
@@ -0,0 +1,194 @@
+#define ADC1_BASE                       (ADC12_BASE + 0x000)
+#define ADC2_BASE                       (ADC12_BASE + 0x100)
+#define ADC_CMN_BASE                    (ADC12_BASE + 0x300)
+#define SDMMC3_BASE                     (PERIPH_BASE_AHB2 + 0x04000)
+#define DLYBSD3_BASE                    (PERIPH_BASE_AHB2 + 0x05000)
+#define USB_OTG_BASE                    (PERIPH_BASE_AHB2 + 0x01000000)
+
+/* AHB3 */
+#define HSEM_BASE                       (PERIPH_BASE_AHB3 + 0x0000)
+#define IPCC_BASE                       (PERIPH_BASE_AHB3 + 0x1000)
+#define HASH2_BASE                      (PERIPH_BASE_AHB3 + 0x2000)
+#define RNG2_BASE                       (PERIPH_BASE_AHB3 + 0x3000)
+#define CRC2_BASE                       (PERIPH_BASE_AHB3 + 0x4000)
+#define CRYP2_BASE                      (PERIPH_BASE_AHB3 + 0x5000)
+#define DCMI_BASE                       (PERIPH_BASE_AHB3 + 0x6000)
+
+/* AHB4 */
+#define RCC_BASE                        (PERIPH_BASE_AHB4 + 0x0000)
+#define PWR_BASE                        (PERIPH_BASE_AHB4 + 0x1000)
+#define GPIO_PORT_A_BASE                (PERIPH_BASE_AHB4 + 0x2000)
+#define GPIO_PORT_B_BASE                (PERIPH_BASE_AHB4 + 0x3000)
+#define GPIO_PORT_C_BASE                (PERIPH_BASE_AHB4 + 0x4000)
+#define GPIO_PORT_D_BASE                (PERIPH_BASE_AHB4 + 0x5000)
+#define GPIO_PORT_E_BASE                (PERIPH_BASE_AHB4 + 0x6000)
+#define GPIO_PORT_F_BASE                (PERIPH_BASE_AHB4 + 0x7000)
+#define GPIO_PORT_G_BASE                (PERIPH_BASE_AHB4 + 0x8000)
+#define GPIO_PORT_H_BASE                (PERIPH_BASE_AHB4 + 0x9000)
+#define GPIO_PORT_I_BASE                (PERIPH_BASE_AHB4 + 0xa000)
+#define GPIO_PORT_J_BASE                (PERIPH_BASE_AHB4 + 0xb000)
+#define GPIO_PORT_K_BASE                (PERIPH_BASE_AHB4 + 0xc000)
+#define EXTI_BASE                       (PERIPH_BASE_AHB4 + 0xd000)
+
+/* APB3 */
+#define SYSCFG_BASE                     (PERIPH_BASE_APB3 + 0x0000)
+#define LPTIM2_BASE                     (PERIPH_BASE_APB3 + 0x1000)
+#define LPTIM3_BASE                     (PERIPH_BASE_APB3 + 0x2000)
+#define LPTIM4_BASE                     (PERIPH_BASE_APB3 + 0x3000)
+#define LPTIM5_BASE                     (PERIPH_BASE_APB3 + 0x4000)
+#define VREFBUF_BASE                    (PERIPH_BASE_APB3 + 0x5000)
+#define SAI4_BASE                       (PERIPH_BASE_APB3 + 0x7000)
+#define DTS_BASE                        (PERIPH_BASE_APB3 + 0x8000)
+#define HDP_BASE                        (PERIPH_BASE_APB3 + 0xa000)
+
+/* AHB5 */
+#define BKPSRAM_BASE                    (PERIPH_BASE_AHB5 + 0x0000)
+#define CRYP1_BASE                      (PERIPH_BASE_AHB5 + 0x1000)
+#define HASH1_BASE                      (PERIPH_BASE_AHB5 + 0x2000)
+#define RNG1_BASE                       (PERIPH_BASE_AHB5 + 0x3000)
+#define GPIO_PORT_Z_BASE                (PERIPH_BASE_AHB5 + 0x4000)
+
+/* AXIM */
+#define AXIMC_BASE                      (PERIPH2_BASE + 0x07000000)
```

Comment:
honestly, I don't really think the PERIPH_BASE and PERIPH2_BASE save us anything or bring any clarity.

Reply:
I followed naming proposed by reference manual §2.5.2, it make it easier to fill in the individual peripherals base address in my opinion.